### PR TITLE
Adding sns_topic module to manage AWS SNS topics

### PR DESCRIPTION
**Pull request by joelthompson**

### CODE REVIEW

**Comment by evanccnyc** [comment]
Good stuff would be a nice add.

**Comment by gregdek** [comment]
@evanccnyc had a chance to test? :)

**Comment by gregdek** [comment]
@joelthompson Thanks for submitting this new module to Ansible Extras! This module is now in community review, a process that is open to all Ansible users. In order for this module to be approved, it must gain the following votes:

“works_for_me”: If you have tested the module thoroughly, including testing of all of the module’s options, and if the module works for you, please add “works_for_me” in the comments.

“passes_guidelines”: If you have gone through the module guidelines and the module meets all of the requirements, please add “passes_guidelines” in the comments. Guidelines are available here: http://docs.ansible.com/developing_modules.html#module-checklist

“needs_revision”: If the module fails to work for you, or if it doesn’t meet guidelines, please add “needs_revision” in the comments with details about what needs to be fixed.

When a module has both “works_for_me” and “passes_guidelines” tags, we will promote the module for inclusion in Ansible Extras.  At this point, you will be expected to maintain the module by fixing bugs and evaluating pull requests in a timely manner. 

Thanks again for submitting your Ansible module!

File: cloud/amazon/sns_topic.py [hunk 1]
```diff
@@ -0,0 +1,261 @@
...
+    default: True
+extends_documentation_fragment: aws
+requirements: [ "boto" ]
+"""
+
+EXAMPLES = """
+
+- name: Create alarm SNS topic
+  sns_topic:
+  name: "alarms"
```
Comment (evanccnyc) [inline cloud/amazon/sns_topic.py:55]: The name, state, display name etc need to be indented once more

File: cloud/amazon/sns_topic.py [hunk 2]
```diff
@@ -0,0 +1,261 @@
...
+          maxDelayTarget: 4
+          numRetries: 3
+          numMaxDelayRetries: 5
+          backoffFunction: "<linear|arithmetic|geometric|exponential>"
+      disableSubscriptionOverrides: True
+      defaultThrottlePolicy: 
+          maxReceivesPerSecond: 10
+  subscriptions:
+    - endpoint: "my_email_address@example.com"
+      protocol: "smtp"
```
Comment (evanccnyc) [inline cloud/amazon/sns_topic.py:71]: protocol is: email

**Comment by evanccnyc** [comment]
@gregdek this works for me so "shipit" once he fixes the example.

**Comment by gregdek** [comment]
Well, sadly, I'm an idiot, and didn't realize that this module was submitted to Core and not Extras -- and we're trying not to accept new modules into Core right now.

So, @joelthompson, if you would do me a favor and reopen it in the Extras repo, that would be greatly appreciated. Be sure to link to this PR in the description, and we can carry over @evanccnyc's +1 comment, which should get us well on the way to approval over there. 

Again, apologies for my brain cramp here.

**Comment by evanccnyc** [comment]
@joelthompson any chance you could resubmit this to ansible-extras? id love to have this included.

**Comment by evanccnyc** [comment]
@joelthompson did you ever resubmit this? It would be useful.

**Comment by joelthompson** [comment]
Hey @evanccnyc -- sorry this took so long, can't go into details but I didn't drop it. Resubmitted to extras, hope you find it useful! :)

**Pull request closed without merging**